syntax = "proto3";
package server;

import "common.proto";

service FrontEnd {
    rpc start_inference(InferenceInfo) returns (InferenceReturn) ;
    rpc result_callback(CallbackInfo) returns (common.Empty) ;
}


message InferenceInfo {
    common.ModelId model_id = 1 ;
    repeated string input_files_paths = 2 ;
}

message InferenceReturn {
    common.ModelId model_id = 1 ;
    repeated string output_files_paths = 2 ;
}

message CallbackInfo {
    common.ModelId model_id = 1 ;
    common.RequestId request_id = 2;
    repeated string output_files_paths = 3 ;
}